### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Viacheslav2024**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -2,5 +2,18 @@
...
+    results = dict()
+
+    def wrapper_cache(*args, **kwargs) -> int:
+        if args in results:
+            print("Getting from cache")
+
+            return results[args]
+
+        result = func(*args, **kwargs)
+        results[args] = result
```
Comment (vasylhnatiuk) [inline app/main.py]:
```suggestion
        results[args] = func(*args, **kwargs)
```

File: app/main.py [hunk 2]
```diff
@@ -2,5 +2,18 @@
...
+        print("Calculating new result")
+
+        return result
```
Comment (vasylhnatiuk) [inline app/main.py]:
```suggestion
        return results[args]
```

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (Viacheslav2024) [reply app/main.py]: Yes, of course! Thanks!

**Review by EdAlekseiev — APPROVED** [review]
Good job

**Pull request closed without merging**